helios-tools/src/main/java/com/spotify/helios/cli/CliMain.java

```diff
@@ -67,17 +76,37 @@ public static void main(final String... args) {
 
   public CliMain(final PrintStream out, final PrintStream err, final String... args)
       throws Exception {
+    this.authProviderFactory = configureAuthProvider();
     this.parser = new CliParser(args);
     this.out = out;
     this.err = err;
     setupLogging();
   }
 
+  private static AuthProvider.Factory configureAuthProvider() {
```

Comment:
I kind of feel that this should actually go in `HeliosClient`. I mean, there's nothing really CLI-specific about it, is there? That way existing `HeliosClient` usages would continue to work without having to worry about doing `setAuthProviderFactory`. Thoughts?